cobra/core/Model.py

```diff
@@ -230,9 +308,85 @@ def add_reactions(self, reaction_list):
 
         self.reactions += reaction_list
 
+        # from cameo ...
+        self._populate_solver(reaction_list)
+
+    def _populate_solver(self, reaction_list, metabolite_list=None):
+        """Populate attached solver with constraints and variables that
+        model the provided reactions.
+        """
+        constraint_terms = AutoVivification()
+        if metabolite_list is not None:
```

Comment:
This line just initializes the Constraint objects (one for each metabolite). Line 385 sets the coefficients of the Constraint expressions (lhs, basically the S-matrix). The reason that the coefficients are not set here is for performance, since the `set_linear_coefficients` method operates directly on the solver and circumvents all sympy operations.